README.md

```diff
@@ -1,50 +1,95 @@
 [![Build Status](https://travis-ci.org/kubernetes-csi/external-provisioner.svg?branch=master)](https://travis-ci.org/kubernetes-csi/external-provisioner)
 
-# Kubernetes external provisioner that works with CSI volumes
+# CSI provisioner
 
-This is an example external provisioner for Kubernetes which provisions using CSI Volume drivers..  It's under heavy development, so at this time README.md is notes for the developers coding.  Once complete this will change to something user friendly.
+The external-provisioner is a sidecar container that dynamically provisions volumes by calling `ControllerCreateVolume` and `ControllerDeleteVolume` functions of CSI drivers. It is necessary because internal persistent volume controller running in Kubernetes controller-manager does not have any direct interfaces to CSI drivers.
 
-# Build
+## Overview
+The external-provisioner is an external controller that monitors `PersistentVolumeClaim` objects created by user and creates/deletes volumes for them. Full design can be found at Kubernetes proposal at [container-storage-interface.md](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/container-storage-interface.md)
 
-```bash
-make csi-provisioner
-```
+## Compatibility
 
-# Test
+This information reflects the head of this branch.
 
-## Start Kubernetes
+| Compatible with CSI Version                                                                | Container Image                | Min K8s Version |
+| ------------------------------------------------------------------------------------------ | -------------------------------| --------------- |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-provisioner | 1.14            |
 
-Run a local kubernetes cluster built from latest master code
+## Feature status
 
-## Run Storage Provider
+Various external-provisioner releases come with different alpha / beta features. Check `--help` output for alpha/beta features in each release.
 
-### Use HostPath drivers
+Following table reflects the head of this branch.
 
-Go to drivers and run:
+| Feature    | Status  | Default | Description                                                                                   |
+| ---------- | ------- | ------- | --------------------------------------------------------------------------------------------- |
+| Topology   | Beta    | On      | [Topology aware dynamic provisioning](https://kubernetes-csi.github.io/docs/topology.html)  (requires kubelet 1.14 on nodes). |
+| Snapshots* | Alpha   | On      | [Snapshots and Restore](https://kubernetes-csi.github.io/docs/snapshot-restore-feature.html). |
 
-```bash
-bin/hostpathplugin --drivername mydriver  --endpoint unix://tmp/csi.sock --nodeid foobar -v=5
-```
+*) There is no special feature gate for snapshots, it is enabled by running [external-snapshotter](https://github.com/kubernetes-csi/external-snapshotter/) and creating its CRDs.
```

Comment:
thinking more about this, maybe we should still have a feature gate?  What if someone tries to use snapshots with a beta/GA driver and it causes external-provisioner to crash?

Reply:
Such change would break existing deployments. There was no alpha feature gate, yet snapshots were enabled before.

BTW, the same is going to happen to migration. It's enabled in kubelet / controller-manager, external-provisioner is just going to provision volumes for in-tree volumes without any gates

Reply:
@davidz627 @ddebroy, FYI ^

Reply:
Given that snapshots is still alpha I think it is fine to break existing deployments (with a release note).  From a support perspective, we don't want to compromise our beta/GA quality with alpha features.

For migration, I think it is fine to have the feature gate controlled by k8s.  Compared to snapshots, there is nothing we can do to prevent users from using the alpha crds

Reply:
Right, I think for migration if the gates are all on in k8s and k8s is requesting that the external provisioner respond to the migrated volumes - if it has the capability it should just do it. It doesn't seem strictly necessary to me to gate the sidecars seperately

Reply:
Thinking about the snapshots case a little more.  Is the snapshots logic in the external-provisioner only triggered by the presence of the PVC.DataSource field?  If so, then that's the same situation as migration, so that's fine.

If you could invoke snapshots logic just from creating the snapshot CRs, then we should feature gate it.